### PR TITLE
fix vllm v0.16.1.dev

### PR DESCRIPTION
**Pull request by mikasenghaas**
fixes the custom token endpoint `v1/chat/completions/token` which broke bc vllm moved around some configs

```bash
uv run rl @ examples/alphabet_sort/rl.toml --clean-output-dir --max-steps 5
```

Screenshot 2026-03-05 at 10 26 00 AM

---

> [!NOTE]
> **Medium Risk**
> Touches the request scheduling/generation path for the token-based chat completions endpoint, so regressions could affect max-token limits or reasoning-termination behavior, but the change is localized to this custom endpoint.
> 
> **Overview**
> Fixes the custom `POST /v1/chat/completions/tokens` flow to match newer vLLM APIs by generating directly from `engine_prompt` + `SamplingParams` rather than building an `engine_request` via `input_processor`.
> 
> The handler now derives `max_model_len` from `model_config.max_model_len`, passes `override_max_tokens` into `get_max_tokens`, computes `reasoning_ended` from extracted prompt token ids, and removes logits-processor validation/imports that no longer apply.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 38e52b895fe06be1a5f9b9b740d3aad2f1d4d69f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/prime_rl/inference/vllm/serving_chat_with_tokens.py [hunk 1]
```diff
@@ -146,31 +149,33 @@ async def create_chat_completion_with_tokens(
...
                         parameter="input_tokens",
                         value=prompt_len,
                     )
 
                 max_tokens = get_max_tokens(
-                    self.max_model_len,
+                    max_model_len,
                     request.max_completion_tokens if request.max_completion_tokens is not None else request.max_tokens,
-                    prompt_len,
+                    self._extract_prompt_len(engine_prompt),
```
Comment (JohannesHa) [inline src/prime_rl/inference/vllm/serving_chat_with_tokens.py:176]: seems unnecessary to call this func again?

**Review by JohannesHa — APPROVED** [review]

File: src/prime_rl/inference/vllm/serving_chat_with_tokens.py (hunk 1 above)
Comment (mikasenghaas) [reply src/prime_rl/inference/vllm/serving_chat_with_tokens.py:176]: copy paste from vllm lol
